### PR TITLE
adding test coverage

### PR DESCRIPTION
**Pull request by hshaosf**
To run test coverage locally:
* python -m pytest --cov=service tests/  --cov-fail-under=100

### CODE REVIEW

**Comment by akegan** [comment]
are you adding a coverage check to circleCI too?

**Comment by hshaosf** [comment]
> are you adding a coverage check to circleCI too?

it is now! 663af05

File: .circleci/config.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ jobs:
           name: run tests
           command: |
             . venv/bin/activate
-            pytest tests
+            python -m pytest --cov=service tests/
```
Comment (akegan) [inline .circleci/config.yml]: when does this fail? any time it goes down?

Comment (akegan) [reply .circleci/config.yml]: I wonder if it automatically fails if test coverage goes down, or if you need to set a threshold explicitly with e.g. 
--cov-fail-under=MIN

Comment (hshaosf) [reply .circleci/config.yml]: thanks! 
its now at e546bf1

**Review by akegan — APPROVED** [review]
looks good! I won't blame you if coverage requirements dip under 100% at some point, but seems like a good start at least!

**Pull request merged**